File: week_2/day_09_monotonic/variations_que/removeKDigits.js
function removeKdigits(num, k) {
    const stack = [];

    for (const d of num) {
        while (stack.length && k > 0 && stack[stack.length - 1] > d) {
            stack.pop();
            k--;
        }
        stack.push(d);
    }

    let res = stack.slice(0, stack.length - k).join("").replace(/^0+/, "");
    return res || "0";
}

File: week_2/day_09_monotonic/variations_que/remove_k_digits.py
def removeKdigits(num, k):
    stack = []

    for d in num:
        while stack and k > 0 and stack[-1] > d:
            stack.pop()
            k -= 1
        stack.append(d)

    result = ''.join(stack[:len(stack)-k]).lstrip('0')
    return result if result else "0"


# INPUT
print(removeKdigits("1432219", 3))
